pallets/capacity/src/benchmarking.rs

```diff
@@ -0,0 +1,122 @@
+use super::*;
+use crate::Pallet as Capacity;
+
+use frame_benchmarking::{account, benchmarks, whitelist_account, Vec};
+use frame_support::{assert_ok, traits::Currency};
+use frame_system::RawOrigin;
+
+const SEED: u32 = 0;
+
+fn assert_last_event<T: Config>(generic_event: <T as Config>::RuntimeEvent) {
+	frame_system::Pallet::<T>::assert_last_event(generic_event.into());
+}
+
+pub fn register_provider<T: Config>(target_id: MessageSourceId, name: &'static str) {
+	let name = Vec::from(name).try_into().expect("error");
+	assert_ok!(T::BenchmarkHelper::create(target_id, name));
+}
+pub fn create_funded_account<T: Config>(
+	string: &'static str,
+	n: u32,
+	balance_factor: u32,
+) -> T::AccountId {
+	let user = account(string, n, SEED);
+	whitelist_account!(user);
+	let balance = T::Currency::minimum_balance() * balance_factor.into();
+	let _ = T::Currency::make_free_balance_be(&user, balance);
+	assert_eq!(T::Currency::free_balance(&user), balance.into());
+	user
+}
+
+// In the benchmarks we expect a new epoch to always start so as to test worst case scenario.
+pub fn set_up_epoch<T: Config>(current_block: T::BlockNumber, current_epoch: T::EpochNumber) {
+	CurrentEpoch::<T>::set(current_epoch);
+	let epoch_start = current_block.saturating_sub(Capacity::<T>::get_epoch_length());
+	CurrentEpochInfo::<T>::set(EpochInfo { epoch_start });
+	CurrentEpochUsedCapacity::<T>::set(450_000u32.into());
+}
+
+benchmarks! {
+	stake {
+		let caller: T::AccountId = create_funded_account::<T>("account", SEED, 105u32);
+		let amount: BalanceOf<T> = T::MinimumStakingAmount::get();
+		let capacity: BalanceOf<T> = Capacity::<T>::calculate_capacity(amount);
+		let target = 1;
+
+		register_provider::<T>(target, "Foo");
+
+	}: _ (RawOrigin::Signed(caller.clone()), target, amount)
+	verify {
+		assert!(StakingAccountLedger::<T>::contains_key(&caller));
```

Comment:
nit: Would be worth asserting any values than just knowing key exists?

Reply:
I think the actual test takes care of those. I sometimes go back and forth on how much we need in here.